src/algorithm.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+// Pragma once is supported on all modern compilers
+// TODO: decide whether to use include guards anyway
```

Comment:
No problem. But just to be pedantic, [which platforms don't support it](http://en.wikipedia.org/wiki/Pragma_once#Portability)?